sycl/include/CL/sycl/intel/sub_group.hpp

```diff
@@ -116,42 +116,38 @@ struct sub_group {
     return __spirv_GroupAll(__spv::Scope::Subgroup, predicate);
   }
 
+
+  template <typename T>
+  using EnableIfIsArithmeticOrHalf = typename std::enable_if<
+      (std::is_arithmetic<T>::value ||
+       std::is_same<typename std::remove_const<T>::type, half>::value),
```

Comment:
Will it work for cl_half or __fp16 data types.

Reply:
@vladimirlaz  This is checking for __fp16 which is half type in SYCL. cl_half is just uint16_t and it is already handled. 

> Will it work for cl_half or __fp16 data types? i think it would be helpfull.